app/src/androidTest/java/com/github/se/studybuddies/tests/GroupsHomeTest.kt

```diff
@@ -8,25 +8,191 @@ package com.github.se.studybuddies.tests
+
+      addLinkButton {
+        assertIsDisplayed()
+        assertHasClickAction()
+        performClick()
+      }
+    }
+  }*/
+
+  @Test
+  fun testDrawerGroup() {
+    ComposeScreen.onComposeScreen<GroupsHomeScreen>(composeTestRule) {
+      drawerScaffold { assertIsDisplayed() }
+      groupsTitle {
+        assertIsDisplayed()
+        assertTextEquals("Groups")
+      }
+      topAppBox { assertIsDisplayed() }
+      topAppBar { assertIsDisplayed() }
+      drawerMenuButton {
+        assertIsDisplayed()
+        assertHasClickAction()
+        performClick()
+      }
+      drawerSheet { assertIsDisplayed() }
+      settingsButton {
+        assertIsDisplayed()
+        assertHasClickAction()
+        performClick()
+      }
+      verify { mockNavActions.navigateTo("${Route.SETTINGS}/${Route.GROUPSHOME}") }
+      confirmVerified(mockNavActions)
+
+      drawerMenuButton {
+        assertIsDisplayed()
+        assertHasClickAction()
+        performClick()
+      }
+      accountButton {
+        assertIsDisplayed()
+        assertHasClickAction()
+        performClick()
+      }
+      verify { mockNavActions.navigateTo("${Route.ACCOUNT}/${Route.GROUPSHOME}") }
+      confirmVerified(mockNavActions)
+    }
+  }
+
+  @Test
+  fun testBottomBarGroups() {
+    ComposeScreen.onComposeScreen<GroupsHomeScreen>(composeTestRule) {
+      groupBottomBar { assertIsDisplayed() }
+
+      soloStudyBottom {
+        assertIsDisplayed()
+        assertHasClickAction()
+        performClick()
+      }
+      verify { mockNavActions.navigateTo(Route.SOLOSTUDYHOME) }
+      confirmVerified(mockNavActions)
+
+      groupsBottom { assertIsDisplayed() }
+
+      messagesBottom {
+        assertIsDisplayed()
+        assertHasClickAction()
+        performClick()
+      }
+      verify { mockNavActions.navigateTo(Route.DIRECT_MESSAGE) }
+      confirmVerified(mockNavActions)
+
+      mapBottom {
+        assertIsDisplayed()
+        assertHasClickAction()
+        performClick()
+      }
+      verify { mockNavActions.navigateTo(Route.MAP) }
+      confirmVerified(mockNavActions)
+    }
+  }
+}
+
+@RunWith(AndroidJUnit4::class)
+class GroupsHomeTest : TestCase(kaspressoBuilder = Kaspresso.Builder.withComposeSupport()) {
+  @get:Rule val composeTestRule = createComposeRule()
+
+  @get:Rule val mockkRule = MockKRule(this)
+  @RelaxedMockK lateinit var mockNavActions: NavigationActions
+
+  val uid = "userTest"
+
+  @Before
+  fun testSetup() {
+    composeTestRule.setContent { GroupsHome(uid, GroupsHomeViewModel(uid), mockNavActions) }
+  }
 
   @Test
-  fun titleAndButtonAreCorrectlyDisplayed() {
-    ComposeScreen.onComposeScreen<com.github.se.studybuddies.screens.GroupsHomeScreen>(
-        composeTestRule) {}
+  fun groupList() {
+    ComposeScreen.onComposeScreen<GroupsHomeScreen>(composeTestRule) {
+      groupBox { assertIsDisplayed() }
+      circularLoading { assertIsDisplayed() }
+      runBlocking { delay(3000) }
```

Comment:
Predefined wait time

Reply:
I'll try to find a solution for this on the next PR where I will realy tests these functionalities